toolkit/tools/internal/packagerepo/repocloner/rpmrepocloner/rpmrepocloner.go

```diff
@@ -587,27 +574,25 @@ func (r *RpmRepoCloner) clonePackage(baseArgs []string, enabledRepoOrder ...stri
 }
 
 func convertPackageVersionToTdnfArg(pkgVer *pkgjson.PackageVer) (tdnfArg string) {
-	tdnfArg = pkgVer.Name
 	// TDNF does not accept versioning information on implicit provides.
 	if pkgVer.IsImplicitPackage() {
 		if pkgVer.Condition != "" {
 			logger.Log.Warnf("Discarding version constraint for implicit package: %v", pkgVer)
 		}
+		tdnfArg = pkgVer.Name
 		return
 	}
 
 	// Treat <= as =
 	// Treat > and >= as "latest"
 	switch pkgVer.Condition {
-	case "<=":
-		logger.Log.Warnf("Treating '%s' version constraint as '=' for: %v", pkgVer.Condition, pkgVer)
-		fallthrough
-	case "=":
-		tdnfArg = fmt.Sprintf("%s-%s", tdnfArg, pkgVer.Version)
-	case "":
-		break
+	case "<=", "<", "=":
+		tdnfArg = fmt.Sprintf("%s %s %s", pkgVer.Name, pkgVer.Condition, pkgVer.Version)
+	case "", ">", ">=":
+		tdnfArg = pkgVer.Name
```

Comment:
I think we can simplify further by:
1. completely removing this case,
2. reverting back the removal of `tdnfArg = pkgVer.Name` at the beginning of this function, and
3. reverting the new `tdnfArg = pkgVer.Name` line inside `if pkgVer.IsImplicitPackage()`.

In short: it looks like the only time we want to use something different than the plain `pkgVer.Name` is for the `"<=", "<", "="` case.

Reply:
Updated

Reply:
See above, I think I've settled on a compromise.